### PR TITLE
Doc changes for making Academy non-virtual

### PR DESCRIPTION
**Pull request by chriselion**
Clean up references to subclassing Academy. These docs will need another pass when we convert to singleton, but they're at least accurate for now.

### CODE REVIEW

File: docs/Learning-Environment-Design.md [hunk 1]
```diff
@@ -66,32 +65,27 @@ information.
...
-* `AcademyStep()` — Prepare the environment for the next simulation step. The
-  base Academy class calls this function before calling any `AgentAction()`
-  methods for the current step. You can use this function to update other
-  objects in the scene before the Agents take their actions. Note that the
-  Agents have already collected their observations and chosen an action before
-  the Academy invokes this method.
-
-See [Academy](Learning-Environment-Design-Academy.md) for a complete list of
-the Academy properties and their uses.
+#### Academy resetting
```
Comment (chriselion) [inline docs/Learning-Environment-Design.md]: Moved this section from docs/Learning-Environment-Design-Academy.md  (now defunct)

File: docs/ML-Agents-Overview.md [hunk 2]
```diff
@@ -139,9 +139,7 @@ organize the Unity scene:
   receives and assigning a reward (positive / negative) when appropriate. Each
   Agent is linked to a Policy.
 - **Academy** - which orchestrates the observation and decision making process.
-  Within the Academy, several environment-wide parameters such as the rendering
```
Comment (chriselion) [inline docs/ML-Agents-Overview.md]: Old cleanup

File: docs/Learning-Environment-Design.md [hunk 3]
```diff
@@ -66,32 +65,27 @@ information.
...
+
+Register with the Academy's OnEnvironmentReset delegate
```
Comment (chriselion) [inline docs/Learning-Environment-Design.md]: Is "register" the right wording here?

Comment (vincentpierre) [reply docs/Learning-Environment-Design.md]: Maybe subscribe?
Should this be an Event or an Action ?

File: docs/Learning-Environment-Design.md [hunk 4]
```diff
@@ -66,32 +65,27 @@ information.
...
+to alter the environment at the start of each episode.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design.md]: Maybe ad some code the show how to setup the reset method?

File: docs/Readme.md [hunk 5]
```diff
@@ -21,7 +21,6 @@
 * [Making a New Learning Environment](Learning-Environment-Create-New.md)
 * [Designing a Learning Environment](Learning-Environment-Design.md)
   * [Agents](Learning-Environment-Design-Agents.md)
```
Comment (vincentpierre) [inline docs/Readme.md]: Kind of feels weird to have just one element indented. I would say rename `Agents` and align it?

File: docs/Migrating.md [hunk 6]
```diff
@@ -10,10 +10,19 @@ The versions can be found in
...
 * Trainer steps are now counted per-Agent, not per-environment as in previous versions. For instance, if you have 10 Agents in the scene, 20 environment steps now corresponds to 200 steps as printed in the terminal and in Tensorboard.
 
 ### Steps to Migrate
 * Multiply `max_steps` and `summary_steps` in your `trainer_config.yaml` by the number of Agents in the scene.
+* If you have a class that inherits from Academy:
+  * If the class didn't override any of the virtual methods and didn't store any additional data, you can just replace the instance of it in the scene with an Academy.
+  * If the class had additional data, create a new MonoBehaviour and store the data on this instead.
+  * If the class overrode the virtual methods, create a new MonoBehaviour and move the logic to it:
+    * Move the InitializeAcademy code to MonoBehaviour.OnAwake
+    * Move the AcademyStep code to MonoBehaviour.Update or FixedUpdate
```
Comment (vincentpierre) [inline docs/Migrating.md]: Just say to FixedUpdate. AcademyUpdate was called every Fixed Update

**Pull request merged**